geode-assembly/src/test/java/org/apache/geode/rest/internal/web/RestServersJUnitTest.java

```diff
@@ -16,42 +16,51 @@
 package org.apache.geode.rest.internal.web;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assume.assumeTrue;
 
-import org.apache.geode.distributed.internal.DistributionConfig;
-import org.apache.geode.test.dunit.rules.ServerStarterRule;
-import org.apache.geode.test.junit.categories.IntegrationTest;
-import org.apache.geode.test.junit.categories.RestAPITest;
 import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
 import org.json.JSONArray;
 import org.junit.BeforeClass;
 import org.junit.ClassRule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
+import org.apache.geode.distributed.internal.DistributionConfig;
+import org.apache.geode.internal.AvailablePort;
+import org.apache.geode.test.dunit.rules.ServerStarterRule;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+import org.apache.geode.test.junit.categories.RestAPITest;
+
 @Category({IntegrationTest.class, RestAPITest.class})
 public class RestServersJUnitTest {
 
   @ClassRule
-  public static ServerStarterRule serverStarter =
-      new ServerStarterRule().withRestService(true).withAutoStart();
+  public static ServerStarterRule serverStarter = new ServerStarterRule().withRestService(true);
+
 
   private static GeodeRestClient restClient;
 
   @BeforeClass
   public static void before() throws Exception {
-    assertThat(serverStarter.getHttpPort()).isEqualTo(DistributionConfig.DEFAULT_HTTP_SERVICE_PORT);
+    assumeTrue(
+        "Default port was unavailable for testing.  Please ensure the testing environment is clean.",
+        AvailablePort.isPortAvailable(DistributionConfig.DEFAULT_HTTP_SERVICE_PORT,
+            AvailablePort.SOCKET));
+    serverStarter.startServer();
     restClient = new GeodeRestClient("localhost", serverStarter.getHttpPort());
```

Comment:
we will need to add this assert statement back as well:

assertThat(serverStarter.getHttpPort()).isEqualTo(DistributionConfig.DEFAULT_HTTP_SERVICE_PORT);

Reply:
We somewhat do this below in lines 66-68, but I can add that assertion explicitly, too.